<launch>
    <node name="carolus_astrobee_rex" pkg="carolus_astrobee" type="carolus_astrobee"  output="screen">
        
        <!-- Execution Parameters -->
        <param name="num_threads" value="4" />
        <param name="queue_size" value="10" />

        <!-- Blob Filtering Parameters -->
        <param name="min_circularity" value="0.4" />
        <param name="saturation_threshold" value="20" />
        <param name="min_area" value="250.0" />
        <param name="max_area" value="2500.0" />
        <param name="max_distance_lim" value="1000.0" />
        <param name="lb_hue" value="160.0" />
        <param name="ub_hue" value="180.0" />

        <!-- Preprocessing Parameters -->
        <param name="kernel_size_gaussian" value="3" />
        <param name="kernel_size_morph" value="3" />
        <param name="image_threshold" value="240" />

        <!-- Camera Parameters -->
        <param name="nav_cam" value="false" />
        <param name="dock_cam" value="false" />
        <param name="sci_cam_compressed" value="true" />
        <param name="sci_cam" value="false" />
        <param name="fisheye" value="false" />
        <param name="fov" value="false" />
        <param name="mono" value="false" />

        <!-- Robot Configuration -->
        <param name="bot_name" value="wannabee" />

        <!-- Camera Intrinsic Parameters -->
        <param name="fx" value="859.44015" />
        <param name="fy" value="849.35466" />
        <param name="cx" value="754.24485" />
        <param name="cy" value="487.7349" /> 

        <!-- Distortion Coefficients  -->
        <param name="distortion" value="0.128628, -0.167456, 0.00213421, -0.00262211" />

        <!-- Topic Names -->
        <param name="dock_cam_topic" value="/d455/color/image_raw" />
        <param name="nav_cam_topic" value="/d455/color/image_raw" />
        <param name="sci_cam_topic" value="/hw/cam_sci" />
        <param name="processed_image_topic" value="/postprocessed/image" />
        <param name="pose_topic" value="/pose" />

        <!-- Benchtest Mode -->
        <param name="benchtest" value="false" />

        <!-- FIFO Filtering -->
        <param name="filter_size" value="5" />
        <param name="translation_threshold" value="10.0" />
        <param name="rotation_threshold" value="10.0" />
        <param name="fifo_on" value="false" />
        <param name="max_time_fifo" value="5.0" />
        <param name="reject_limit" value="40" />

        <!-- Known Beacon Points -->
        <param name="known_points" value="0.055 0.0 0.0 -0.055 0.0 0.0 0.0 0.048 0.0 0.0 0.0 0.037" />

    </node>
</launch>


